sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -489,13 +511,13 @@ object BooleanSimplification extends Rule[LogicalPlan] with PredicateHelper {
         case (l, Or(r, l1)) if (Not(l) == l1) => And(l, r)
         case (Or(l, l1), r) if (l1 == Not(r)) => And(l, r)
         case (Or(l1, l), r) if (l1 == Not(r)) => And(l, r)
-        // (a || b) && (a || c)  =>  a || (b && c)
+        // (a || b) && (a || b || c)  =>  a || b
```

Comment:
`(a || b) && (a || c)  =>  a || (b && c)` is just a transformation instead of optimization, it is only the case when we could eliminate one side like: `(a || b) && (a || b || c)  =>  a || b`.  Besides, the original transformation is opposite to CNF Normalize.